osf/models/sanctions.py

```diff
@@ -484,22 +562,12 @@ def _email_template_context(self,
             })
         return context
 
-    def reject(self, user, token):
-        reg = self._get_registration()
-        if reg.is_public:
-            raise HTTPError(
-                http_status.HTTP_400_BAD_REQUEST,
-                data={
-                    'message_short': 'Registrations cannot be modified',
-                    'message_long': 'This project has already been registered and cannot be deleted',
-                }
-            )
```

Comment:
This logic is now handled as an unsupported State Machine transition (which are caught and converted into an HTTP Error universally in the SanctionsStateMachine class)